### PR TITLE
update include_class

### LINKED ISSUE #1236: Where should i modify when training some of parts of classes in custom dataset.
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I wondered where should i modify in Yolov8, like this [issue](https://github.com/ultralytics/yolov5/issues/1978)(referenced in yolov5)

I wanna train specific classes in my dataset.
```
0 0.00475594300630158 0.7151455927652908 0.9992383128629915 0.7151455927652908 0.9992383128629915 0.8626347866854197 0.00475594300630158 0.8626347866854197
1 0.8647255613720259 0.018674381034945398 0.9346355586715483 0.018674381034945398 0.9346355586715483 0.9794143971359162 0.8647255613720259 0.9794143971359162
1 0.9351224325517384 0.017900524229638153 0.9999091168756978 0.017900524229638153 0.9999091168756978 0.9801048459276309 0.9351224325517384 0.9801048459276309
4 0.00485575549842902 0.8646976090014065 0.5845004875454304 0.8646976090014065 0.5845004875454304 0.9766041030117855 0.00485575549842902 0.9766041030117855
3 0.004968082955431675 0.023296828331673985 0.9985846740417664 0.023296828331673985 0.9985846740417664 0.1598350163561371 0.004968082955431675 0.1598350163561371
2 0.18670483460559797 0.003551696921862668 0.2926844783715013 0.004498816101026046 0.29497455470737916 0.9949486977111286 0.18848600508905855 0.9947119179163377
```
There are 5 classes in my custom data, but i wanna train only "1, 2" class.
Could you please tell me the way of modify it??

### Additional

_No response_

### PR DESCRIPTION
**Pull request by Laughing-q**
@glenn-jocher @yellowjs0304
now `include_class` can be used by modifying this line:
[https://github.com/ultralytics/ultralytics/blob/93b5f9600ae862042eabec0d6202514afd2e2f5d/ultralytics/yolo/data/base.py#L48](https://github.com/ultralytics/ultralytics/blob/93b5f9600ae862042eabec0d6202514afd2e2f5d/ultralytics/yolo/data/base.py#L48)
just add the class idx you want to train.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved class index verification and refined label filtering logic.

### 📊 Key Changes
- Added a check to ensure valid class indices in dataset YAML.
- Moved `_apply_default_class_names` from an instance method to a static method within `AutoBackend`.
- Modified the dataset loading process to allow explicit inclusion of classes using the `classes` parameter instead of just `single_cls`.
- Updated label update mechanism to properly handle segment filtering based on included classes.
- Ensured class names are checked and converted from lists or tuples to dictionaries within the dataset check function.

### 🎯 Purpose & Impact
- Ensures dataset integrity by enforcing correct class index ranges, preventing confusion and errors during model training. 🧐
- Makes dataset class filtering more flexible, allowing for selective class training and analysis, which can be useful for targeted model training scenarios. 🎯
- Simplifies the logic of label and segment processing, potentially enhancing the robustness and readability of the codebase. 📖
- Integrates dataset checks for class name formats more tightly into the data loading process, increasing reliability when working with different dataset configurations. 👍

These updates aim to improve usability for those creating custom datasets and training models with selective classes, as well as maintain code quality and reduce potential sources of errors.

### CODE REVIEW

**Comment by Laughing-q** [comment]
@glenn-jocher do you think we need a arg for this? i.e `include_classes`

**Comment by glenn-jocher** [comment]
@Laughing-q oh I know what we should do, we should connect this to the existing `classes` argument. During predict/val it fliters detections by class, only showing the classes you pass. We can have it do the same for training :)

**Comment by Laughing-q** [comment]
@glenn-jocher got it!

**Comment by Laughing-q** [comment]
@glenn-jocher done! BTW I updated the condition to `if include_class is not None` otherwise it won't work properly if the filter class is `classes=0`.
usage:
```bash
yolo detect train classes=0  # classes=0,1,... for multiple classes
```
```python
model.train(classes=0)   # classes=[0,1,2...,...] for multiple classes
```

**Comment by glenn-jocher** [comment]
@Laughing-q nice work! PR merged into updates for 8.0.50 today.
